src/commcare_cloud/ansible/group_vars/all.yml

```diff
@@ -54,8 +54,8 @@ old_s3_blob_db_s3_bucket: 'blobdb'
 
 # To use these vars without deploying the riak control machine
 # add riakcs_s3_access_key and riakcs_s3_secret_key to the ansible secret config directory
-s3_blob_db_access_key: "{{ riakcs_s3_access_key if riakcs_s3_access_key else (hostvars[riakcs_control]['riak_key'] if s3_blob_db_enabled else '') }}"
-s3_blob_db_secret_key:  "{{ riakcs_s3_secret_key if riakcs_s3_secret_key else (hostvars[riakcs_control]['riak_secret'] if s3_blob_db_enabled else '') }}"
+s3_blob_db_access_key: "{{ secrets.S3_BLOB_DB_ACCESS_KEY | default(None) }}"
+s3_blob_db_secret_key:  "{{ secrets.S3_BLOB_DB_SECRET_KEY | default(None) }}"
```

Comment:
doesn't this already use the value from vault if it exists? In `all.yml` there is this:
```
riakcs_s3_access_key: "{{ secrets.RIAKCS_S3_ACCESS_KEY | default(None) }}"
```

Reply:
I think @snopoke is correct, this change is not necessary.

Reply:
I think you should write `secrets.RIAKCS_S3_ACCESS_KEY` and `secrets.RIAKCS_S3_SECRET_KEY` in the vault rather than `secrets.S3_BLOB_DB_ACCESS_KEY` and `secrets.S3_BLOB_DB_SECRET_KEY`. They will be assigned to the proper variable in `all.yml`:

https://github.com/dimagi/commcare-cloud/blob/df9ab5d1fc1705c92ebe617fbf26dee8ce658090/src/commcare_cloud/ansible/group_vars/all.yml#L171-L172

Reply:
Offlined with Daniel that keeping both `s3_blob_db_secret_key` and `riakcs_s3_access_key ` is more robust for within an ansible session (mostly writing this here for my own memory). Updated here: https://github.com/dimagi/commcare-cloud/pull/2038/commits/21ffe5dbc4feafdfc1eae0c4deba3aab58189868